pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -54,230 +56,179 @@ func (e transientError) Error() string {
-		if trigger.Type != deployapi.DeploymentTriggerOnImageChange || trigger.ImageChangeParams == nil || !trigger.ImageChangeParams.Automatic {
-			continue
-		}
-		name := trigger.ImageChangeParams.From.Name
-		tag := trigger.ImageChangeParams.Tag
-		istag := imageapi.JoinImageStreamTag(name, tag)
-
-		// Check if the image stream tag pointed by the trigger exists
-		if _, err := c.osClient.ImageStreamTags(config.Namespace).Get(name, tag); err != nil {
-			if !errors.IsNotFound(err) {
-				glog.V(2).Infof("Error while trying to get image stream tag %q: %v", istag, err)
-				return details, existingDeployments, latestDeploymentExists, nil
+	activeDeployment := deployutil.ActiveDeployment(config, existingDeployments)
+	// Compute the replica count for the active deployment (even if the active
+	// deployment doesn't exist). The active replica count is the value that
+	// should be assigned to the config, to allow the replica propagation to
+	// flow downward from the config.
+	//
+	// This takes into account resources predating the propagation behavior
+	// change, as well as external modifications to the deployments (e.g.
+	// scalers).
+	activeReplicas := config.Template.ControllerTemplate.Replicas
+	if activeDeployment != nil {
+		activeDeploymentIsLatest := activeDeployment.Name == latestDeployment.Name
+		lastActiveReplicas, hasLastActiveReplicas := deployutil.DeploymentReplicas(activeDeployment)
+		if activeDeploymentIsLatest {
+			if !hasLastActiveReplicas || lastActiveReplicas != activeDeployment.Spec.Replicas {
+				activeReplicas = activeDeployment.Spec.Replicas
 			}
-			// In case the image stream tag was not found, then it either doesn't exist or doesn't exist yet
-			// (a build configuration output points to it so it's going to be populated at some point in the
-			// future)
-			details = fmt.Sprintf("The image trigger for image stream tag %q will have no effect because image stream tag %q does not exist.", istag, istag)
-			bcList, err := c.osClient.BuildConfigs(kapi.NamespaceAll).List(labels.Everything(), fields.Everything())
-			if err != nil {
-				glog.V(2).Infof("Error while trying to list build configs: %v", err)
-				return details, existingDeployments, latestDeploymentExists, nil
-			}
-
-			for _, bc := range bcList.Items {
-				if bc.Spec.Output.To != nil && bc.Spec.Output.To.Kind == "ImageStreamTag" {
-					parts := strings.Split(bc.Spec.Output.To.Name, ":")
-					if len(parts) != 2 {
-						glog.V(2).Infof("Invalid image stream tag: %q", bc.Spec.Output.To.Name)
-						return details, existingDeployments, latestDeploymentExists, nil
-					}
-					if parts[0] == name && parts[1] == tag {
-						details = fmt.Sprintf("The image trigger for image stream tag %q will have no effect because image stream tag %q does not exist.\n\tIf image stream tag %q is expected, check build config %q which produces image stream tag %q.", istag, istag, istag, bc.Name, istag)
-						break
-					}
+		} else {
+			if hasLastActiveReplicas {
```

Comment:
I just redid all the test cases, so you'll need to go through and help me vet each case. If you think you found a new edge case we haven't accounted for, please whip up a new scenario to prove it's broken (or explain how the test scenario is broken, etc.)

Thanks for the help!